macosx/kext/AudioDevice.cpp

```diff
@@ -757,15 +824,39 @@ IOReturn kXAudioDevice::gainChanged(IOAudioControl *gainControl, SInt32 oldValue
 {
     debug(DBGCLASS"[%p]::gainChanged(%p, %ld, %ld)\n", this, gainControl, oldValue, newValue);
     
-    if (gainControl) {
-        debug("\t-> Channel %ld\n", gainControl->getChannelID());
-    }
+//    if (gainControl) {
+//        debug("\t-> Channel %ld\n", gainControl->getChannelID());
+//    }
     
-    // Add hardware gain change code here 
+    // Add hardware gain change code here
+    if (gainControl && prolog_pgm!=-1){
+        dword gain;
+        int id = gainControl->getChannelID();
+        gain=(dword)((newValue+1)*(0x2000));
+        
+        if(id == 1){
```

Comment:
Why not to use kIOAudioControlChannelIDDefaultLeft and similar constants here?

Reply:
It's clearer than kIOAudioControlChannelIDDefaultRightRear for input.

Reply:
Ah, yes, ok, I agree. Let's use plain numbers for recording channels.